### PR TITLE
Removed Guava, bump Gradle and all deps.

### PR DESCRIPTION
**Pull request by soberich**
Hi,
We really want to remove Guava as it is on of the largest deps in war though required by this dependency (and json-patch). This would ease the choice to depend or not on Guava.
Please kindly review.
No "optimizations" made. Just a plain migration with practically coping and pasting logic from Guava's methods (Guava is a great library but we just have a JAX-RS web-app and Guava is a too big contributor to classpath).

Overview.

Removed Guava
Bump gradlew to 5.0 and adjust scripts for compatibility
Builds and tests run now on Java 6-11
Bump all versions to top available.

### CODE REVIEW

**Comment by ar** [comment]
There's also [CVE-2018-10237](https://nvd.nist.gov/vuln/detail/CVE-2018-10237) that hits 16.0.1.

If this PR is not accepted, would you consider using Guava `27.1-jre` instead of 16.0.1 ?

**Comment by Capstan** [comment]
I am starting to look at CVEs in this and its dependents. I want to sever the Gradle changes from the Guava changes, so I'll probably crib from this to get it building, commit, and then evaluate code changes separately. /cc @huggsboson

**Comment by Capstan** [comment]
Guava has been updated in pr #8 so the Guava removal can be considered on its own.

**Comment by soberich** [comment]
@Capstan Guys, wait. What is the reason to leave Guava. Again, let's remove it. It is not needed. I see this as simply much better approach. Why do half-action when we should go. @Capstan Why making PR https://github.com/java-json-tools/jackson-coreutils/pull/8?

**Comment by Capstan** [comment]
Bumping the version was IMO, the safest way to mitigate the CVE that @ar mentioned quickly. It is not meant to represent a final decision on the "whether Guava" front.

**Comment by Capstan** [comment]
@soberich If you rebase and squash your change, we can take a look at it isolation. Thanks!

**Comment by soberich** [comment]
I will. Can we also jackpot this to `fge` coordinates afterwords? This woul be really really in demand I believe.

File: src/main/java/com/github/fge/jackson/JsonNumEquals.java [hunk 1]
```diff
@@ -41,21 +41,50 @@
  * kind of equality.</p>
  */
 public final class JsonNumEquals
-    extends Equivalence<JsonNode>
 {
```
Comment (Capstan) [inline src/main/java/com/github/fge/jackson/JsonNumEquals.java]: Removing this extension seems like a breaking change.

File: src/main/java/com/github/fge/jackson/jsonpointer/JsonPointerException.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 public final class JsonPointerException
     extends Exception
 {
+    static final long serialVersionUID = -1835450876546372005L;
```
Comment (Capstan) [inline src/main/java/com/github/fge/jackson/jsonpointer/JsonPointerException.java]: Out of curiosity, where did this come from? Does this allow `JsonPointerException`s serialized in the old system to be deserialized in the new system?

Comment (soberich) [reply src/main/java/com/github/fge/jackson/jsonpointer/JsonPointerException.java]: It was long ago, but I think it was just a quick refactor for Sonar satisfaction.

File: src/main/java/com/github/fge/jackson/JsonNumEquals.java (hunk 1 above)
Comment (soberich) [reply src/main/java/com/github/fge/jackson/JsonNumEquals.java]: Why?
Do you believe anyone relies in it in hierarchy while working with `JsonNumEquals`?

Comment (Capstan) [reply src/main/java/com/github/fge/jackson/JsonNumEquals.java]: I don't believe anything either way. I do know I cannot survey all users of the library. I do know it's part of the surface, and any user could have legitimately started relying on it. By semver conventions, that implies removing it is a breaking change and would necessitate a major version bump.

I'm also not averse to major version bumps for such changes, but I'm apparently an outlier, as others want the major version to be more associated with a collection of features than with API compat. As the new kid on this block, I'm going to defer to the more senior maintainers, if they express a preference.

Comment (soberich) [reply src/main/java/com/github/fge/jackson/JsonNumEquals.java]: Let me start looking at it. I will today/tomorrow.

Comment (Capstan) [reply src/main/java/com/github/fge/jackson/JsonNumEquals.java]: No rush. Just to expectation set, as soon as I have permissions, I'm going to cut a 1.10 release with just the dependency version bumps. If this commit is accepted, it'll be in a following release.

Comment (soberich) [reply src/main/java/com/github/fge/jackson/JsonNumEquals.java]: @Capstan I am confused a bit. But I didn't remove that I shaded it? Which changes were you looking at. Basically it was all good and you consern already was resolved originally. I will squash it.

**Comment by soberich** [comment]
@Capstan 
I have rebased. 
I have seen this https://github.com/java-json-tools/jackson-coreutils/issues/11, but I don't see Guava now in a dependency tree of my fork (this PR). Maybe I didn't get what you mean there.
Anyway now we have a green bar here to proceed.
Thanks.

**Comment by soberich** [comment]
@Capstan ping.

File: src/main/java/com/github/fge/jackson/JsonNumEquals.java (hunk 1 above)
Comment (Capstan) [reply src/main/java/com/github/fge/jackson/JsonNumEquals.java]: I also probably need to make sure that at least the dependent projects in the java-json-tools don't rely on this. Not sure exactly best way to do that without using snapshot releases.

**Comment by soberich** [comment]
All true. Well, the request was waiting for review and feedback, and as now we are there, we may get to a key question -
Are going to bump the (major) version or not?Because if yes, I will simply properly shadow those classes by  prefixing "com.google..." package name with library name-space.
Initially PR was "for discussion" in my mind on this matter, as if maybe we would want to somehow keep the version.
@Capstan about your previous comment. If we will bump the version I will change package. This should be a solution. Right?

**Comment by soberich** [comment]
@Capstan this should work now. I prefixed as posted in previous commit.
I can squash all after you review if needed.

**Comment by Capstan** [comment]
I screwed up CI. Give me a second.

**Comment by Capstan** [comment]
Ah, no, CI should be fixed as of commit ce6f3a8. Sync and try again?

**Comment by soberich** [comment]
@Capstan WDYT?

**Comment by soberich** [comment]
@Capstan I rebased it multiple times, so now you probably proceed with some javadoc linting and javadoc fails. Can we finish the work and merge. You can may minor changes youself such as javadoc, I it is hard to follow those minors and rebasing everytime. The PR intention is to remove Guava which is done. Let's do it then. Again feel free to make any adjustments.

**Comment by soberich** [comment]
@Capstan just a ping

**Comment by Capstan** [comment]
I committed most of this in #42.

**Pull request closed without merging**